src/Microsoft.VisualStudio.Composition/Reflection/SkipClrVisibilityChecks.cs

```diff
@@ -0,0 +1,142 @@
+            Requires.NotNull(moduleBuilder, nameof(moduleBuilder));
+            this.assemblyBuilder = assemblyBuilder;
+            this.moduleBuilder = moduleBuilder;
+        }
+
+        /// <summary>
+        /// Ensures the CLR will skip visibility checks when accessing
+        /// the assembly that contains the specified member.
+        /// </summary>
+        /// <param name="memberInfo">The member that may not be publicly accessible.</param>
+        internal void SkipVisibilityChecksFor(MemberInfo memberInfo)
+        {
+            this.SkipVisibilityChecksFor(memberInfo.Module.Assembly);
+        }
+
+        /// <summary>
+        /// Add an attribute to the dynamic assembly so that the CLR will skip visibility checks
+        /// for the specified assembly.
+        /// </summary>
+        /// <param name="assembly">The assembly to skip visibility checks for.</param>
+        private void SkipVisibilityChecksFor(Assembly assembly)
+        {
+            Requires.NotNull(assembly, nameof(assembly));
+            var assemblyName = assembly.GetName();
+            this.SkipVisibilityChecksFor(assemblyName);
+        }
+
+        /// <summary>
+        /// Add an attribute to the dynamic assembly so that the CLR will skip visibility checks
+        /// for the assembly with the specified name.
+        /// </summary>
+        /// <param name="assemblyName">The name of the assembly to skip visibility checks for.</param>
+        private void SkipVisibilityChecksFor(AssemblyName assemblyName)
+        {
+            Requires.NotNull(assemblyName, nameof(assemblyName));
+
+            string assemblyNameArg = assemblyName.Name;
+            if (this.attributedAssemblyNames.Add(assemblyNameArg))
+            {
+                var cab = new CustomAttributeBuilder(this.GetMagicAttributeCtor(), new object[] { assemblyNameArg });
+                this.assemblyBuilder.SetCustomAttribute(cab);
+            }
+        }
+
+        /// <summary>
+        /// Gets the constructor to the IgnoresAccessChecksToAttribute, generating the attribute if necessary.
+        /// </summary>
+        /// <returns>The constructor to the IgnoresAccessChecksToAttribute.</returns>
+        private ConstructorInfo GetMagicAttributeCtor()
+        {
+            if (this.magicAttributeCtor == null)
+            {
+                var magicAttribute = this.EmitMagicAttribute();
+                this.magicAttributeCtor = magicAttribute.GetConstructor(new Type[] { typeof(string) });
+            }
+
+            return this.magicAttributeCtor;
+        }
+
+        /// <summary>
+        /// Defines the special IgnoresAccessChecksToAttribute type in the <see cref="moduleBuilder"/>.
+        /// </summary>
+        /// <returns>The generated attribute type.</returns>
+        private TypeInfo EmitMagicAttribute()
+        {
+            var tb = this.moduleBuilder.DefineType(
+                "System.Runtime.CompilerServices.IgnoresAccessChecksToAttribute",
```

Comment:
Okay - that should be fine in VS since we require .NET 4.6. I was initially worried about the situations where we launch `devenv /updateConfiguration` when only .NET 4.5 is installed - this can happen if we're pending reboot to install .NET 4.6. However, I just remember this was "fixed" in 15.2, though, by avoiding composing MEF in the cases where .NET 4.6 is not installed (DevDiv PR 66331). The next launch of VS will do the composition in that case.